Porting the vanilla throwing overhaul

sean-c-johnston opened the pull request:
Removes needles and converts them to darts, removing the need for blowguns. Removes the sleep, paralysis and confuse brand needles/darts and replaces with a new brand that blinds/confuses. Needle/dart brands are renamed - possible brands are now poison, curare, atropa (new blinding brand), and datura (renamed frenzy brand).

Also changes Tomahawks to Boomerangs and makes them always returning. Javelins now always penetrate. Removes other tomahawk/boomerang and javelin brands.

### Testing
Started a game pre-change, on a SpAs. 

1. Confirmed starting with blowgun and needles. 
2. Spawned a ton of Sonja in wizmode, killed a few so they dropped blowguns.
3. Built the new version, reloaded save. Needles in inventory and on the floor have become darts.

Blowguns exist in loot of the Sonjas spawned in the previous build, so any blowguns in piles still exist and any monsters spawned in the old build may still drop them, but the blowguns themselves don't do anything and don't crash the game when wielded, and neither does quivering darts with/without a blowgun wielded. 

Monsters such as Sonja spawned in the new version don't have blowguns in their loot tables, and their needles are replaced with darts.

sdynet committed:
Throwing Overhaul III: Remove most tomahawk and javelin brands

Now that blowguns are removed, throwing users do not need three ways to
poison a monster.

Steel is removed; silver now gets both bonuses, to be a "rare extra
damage throwable" that fits a niche not available with other weapon
types. This might be too good, which can be addressed in a couple of
ways.

Exploding, while "neat" was rarely different from yet another bonus
damage throwable.

Returning and penetration are removed, these effects will return in a
different way in subsequent commits.

[Committer's note: Closes #1048. Significant expansion of the original
 patch]

ebering committed:
Throwing Overhaul I: Remove blowguns, needles become darts

The presence of a *launcher* modified by throwing skill is a wart left
over from the 0.6 ranged reform. The rarity of blowguns limited the use
of needles early on to some extent, the balance impact of this will be
addressed in a subsequent needle-brand-consolidation commit. In exchange
for this limit, the interaction of blowgun enchantment with success
chance made finding an enchanted blowgun a 0-skill no brainer: it was
always better to use it.

This commit removes blowguns. Needles are replaced with darts which are
thrown. Darts have the same brands as needles. To replace blowgun
enchantment, hex dart succes scales as (2/3) * (throwing + stealth),
preserving the ability to land hex darts against high HD targets, but
requiring a strategic cost to do so.

As a bonus, no more stories about bladders or bellows for mummies are
required.

ebering committed:
Throwing Overhaul II: Darts brand consolidation

The hex darts (paralysis, confuse, and sleep) are redundant in several
ways: mutually as a way for needles to generate stabs, with hexes, and
with wands. The result is a clutter of darts.

This commit removes these brands, replacing them with a single new
brand: blinding. In addition to blinding the target for a long duration
(depending on throwing skill) this brand also confuses the target for a
shorter duration. The effect was chosen to be distinct from hexes
(compared to dazzling spray's 4-8 turn blindness and invisibility) and
hex wands.

ebering committed:
New dart tiles (CanOfWorms)

ebering committed:
Rename frenzy and blinding darts

This commit is separate as renames tend to be controversial. All changes
are cosmetic.

Darts of frenzy become "datura-tipped darts" after the genus of toxic
plant that can cause psychosis and darts of blinding become
"atropa-tipped darts" after the genus of toxic plant that can cause
visual distortion and confusion.

Players aren't likely to know this (unless they're a botanist), but they
probably also don't know that curare is (in addition to being a generic
name for blowgun poisons) a genus of toxic plant. So it fits the
"exotic" dart poisons into a nice family.

ebering committed:
Remove some unneeded TAG_MAJORs

ebering committed:
Throwing Overhaul IV: Returning and Penetration

To differentiate the different types of throwables, javelins now always
penetrate. This overlaps some with crossbows, which provide a
penetrating brand, but a crossbow user can choose from many brands and
has access to weapon enchantment.

Tomahawks always return. This is more a QoL change than anything, and a
step towards a broader ranged reform. In terms of ammo availability, the
mulch rate remains the same, so throws per tomahawk at on strategic
timescales are static. On tactical timescales, a player relying on
tomahawks for damage will for the most part not be limited by ammo; a
player 0-skill throwing tomahawks will get a few extra throws in the
early game, but the damage from these extra throws isn't that much.
Since this effect applies to monsters also, giving them a comparative
buff, it should be a wash.

ebering committed:
Throwing Overhaul V: Save Compatibility

Old needles are converted to darts. Old brands are stripped or converted
according to the following table.

|  old brand  | new brand |
| ----------- | --------- |
| penetration | none      |
| returning   | none      |
| exploding   | none      |
| poison*     | none      |
| steel       | silver    |
| sleep       | atropa    |
| paralysis   | atropa    |
| confusion   | atropa    |

* poison is removed from boomerangs and javelins, darts retain it

ebering committed:
Throwing Overhaul VI: Adjust brand weights

Based on objstat runs, the initial weights for the new silver brand was
too high, as was the relative frequency of datura darts.

ebering committed:
Boomerang item tiles (CanOfWorms)

Thrown boomerangs are animated on the beam path.

ebering committed:
Add sdynet to CREDITS.txt

araganzarcrawl committed:
Initial Jiyva Changes

changed chance to eat offlevel item from 1 in 25 to 1 in 500
reduced piety cost to cure bad mutation from 15 to 10
changed minimum hp and mp healing to 2 instead of 1
removed something about jiyva not allowing artefacts to be created
give innate rank of evolution on joining jiyva
make evolution non-innate leaving jiyva
evolution will not remove itself if you worship Jiyva
doubled chance of gift activating after timeout
reduced gift timeout from 15+2d4 to 8+2d3
GIFTS - cut chance of removing mutations from 5%/25% to 2%/8%
GIFTS - added 5% chance to remove a bad mutatoin
GIFTS - changed chance of random/slime/good mut to 20/25/30

araganzarcrawl committed:
Merge pull request #98 from Floodkiller/jiyva_reform

Initial Jiyva Changes

araganzarcrawl committed:
Update religion.cc

change you.mutate to mutate

araganzarcrawl committed:
Merge pull request #99 from Floodkiller/jiyva_reform

Update religion.cc

araganzarcrawl committed:
Update religion.cc

fix to excommunication

araganzarcrawl committed:
Merge pull request #100 from Floodkiller/jiyva_reform

Update religion.cc

araganzarcrawl committed:
Update religion.cc

araganzar is bad at c++

araganzarcrawl committed:
Merge pull request #101 from Floodkiller/jiyva_reform

Update religion.cc

araganzarcrawl committed:
Update timed-effects.cc

replaced god check with you_worship which I hope actually works

araganzarcrawl committed:
Merge pull request #102 from Floodkiller/jiyva_reform

Update timed-effects.cc

araganzarcrawl committed:
Update ng-setup.cc

try to add evolution mutation and give a free slime on slime apostle start

araganzarcrawl committed:
Merge pull request #103 from Floodkiller/jiyva_reform

Update ng-setup.cc

araganzarcrawl committed:
Update ng-setup.cc

don't check has jelly we know dey not has jelly

araganzarcrawl committed:
Merge pull request #104 from Floodkiller/jiyva_reform

Update ng-setup.cc

araganzarcrawl committed:
Wand, Ds mut changes

araganzarcrawl committed:
Merge pull request #105 from Floodkiller/jiyva_reform

Wand, Ds mut changes

araganzarcrawl committed:
Update ng-setup.cc

dont try to spawn jelliez

araganzarcrawl committed:
Merge pull request #106 from Floodkiller/jiyva_reform

Update ng-setup.cc

araganzarcrawl committed:
Update ng-setup.cc

try same method other new game setup uses for species mutations to add innate evolution

araganzarcrawl committed:
Merge pull request #107 from Floodkiller/jiyva_reform

Update ng-setup.cc

araganzarcrawl committed:
Update religion.cc

add more descriptive comments to jiyva mutation actions

araganzarcrawl committed:
Merge pull request #108 from Floodkiller/jiyva_reform

Update religion.cc

araganzarcrawl committed:
Merge pull request #109 from Floodkiller/gooncrawl

pull fixes into goondev

araganzarcrawl committed:
infinite mutation station

take the brakes off

araganzarcrawl committed:
Merge pull request #110 from Floodkiller/jiyva_reform

infinite mutation station

araganzarcrawl committed:
Merge pull request #111 from Floodkiller/gooncrawl

Get fix for security something

seanjohnston-storefeeder committed:
Rename Tomahawks Boomerangs

Now that they always return. And besides, the new name is significantly
more Australian.

Floodkiller commented on crawl-ref/source/dat/des/builder/alphashops.des:
```diff
@@ -81,7 +81,7 @@
        "book of dreams", "book of the dragon", "dagger", "demon blade",
        "demon trident", "demon whip", "dire flail", "double sword",
        "ring of dexterity", "manual of dodging", "wand of disintegration",
-       "disc of storms", "staff of death", "wand of digging" }
```
Disc of storms is being removed here when it shouldn't be.

Floodkiller commented on crawl-ref/source/dat/des/sprint/arena_sprint.des:
```diff
@@ -436,9 +436,9 @@ function arena_sprint_get_monster_set(round, boss)
                 -- Gimmicks
                 "neqoxec", "smoke demon", "catoblepas",
                 "moth of wrath / w:20 stone giant", "ball lightning",
-                "boulder beetle", "orb spider",
```
Boulder beetles are being removed here when they shouldn't.

Floodkiller commented on crawl-ref/source/dat/descript/items.txt:
```diff
@@ -95,11 +95,6 @@ battleaxe
-
-A long, light tube, open at both ends, through which various types of needles
-are propelled by puffs of breath at enemies. It makes very little noise.
-%%%%
```
Does removing this description break anything when examining blowguns that still exist in a save game?

sean-c-johnston replied:
No but it leaves an ugly error message where the description should be. Restored it.

Floodkiller commented on crawl-ref/source/tag-version.h:
```diff
@@ -221,6 +221,14 @@ enum tag_minor_version
+    TAG_MINOR_GAUNTLET_TRAPPED,    // It was briefly possible to get trapped in a specific gauntlet map.
+    TAG_MINOR_REMOVE_DECKS,        // Decks are no more
+    TAG_MINOR_GAMESEEDS,           // Game seeds + rng state saved
+    TAG_MINOR_YELLOW_DRACONIAN_RACID, // Change yellow draconians' rAcid fake mutation to a true mutation.
```
Don't add these tags, we haven't merged the commits associated with them.

sean-c-johnston committed:
remove tags pulled in from vanilla that we haven't merged yet.

sean-c-johnston committed:
re-add some removed items/monsters

sean-c-johnston committed:
restore Blowgun description to avoid ugly error message when inspecting one.

sean-c-johnston committed:
fix typo

sean-c-johnston committed:
Gooncrawl vanilla throwing overhaul (#114)

Porting throwing overhaul from vanilla into Gooncrawl.

Co-authored-by: Aidan Holm <aidanholm+github@gmail.com>
Co-authored-by: Floodkiller <floodkiller3228@gmail.com>
Co-authored-by: Edgar A. Bering IV <trizor@gmail.com>
Co-authored-by: sdynet <sdynet76@gmail.com>
Co-authored-by: seanjohnston-storefeeder <45562757+seanjohnston-storefeeder@users.noreply.github.com>

sean-c-johnston committed:
Gooncrawl vanilla monster item identification (#115)

Porting monster items being identified upon seeing the monster from vanilla into Gooncrawl.

Co-authored-by: Aidan Holm <aidanholm+github@gmail.com>
Co-authored-by: Floodkiller <floodkiller3228@gmail.com>
Co-authored-by: Edgar A. Bering IV <trizor@gmail.com>
Co-authored-by: gammafunk <gammafunk@gmail.com>
Co-authored-by: seanjohnston-storefeeder <45562757+seanjohnston-storefeeder@users.noreply.github.com>

seanjohnston-storefeeder committed:
Fix crash when firing stropa or datura darts.

sean-c-johnston committed:
Merge branch 'gooncrawl_vanilla-throwing-overhaul' of github.com:SurpriseTRex/crawl into gooncrawl_vanilla-throwing-overhaul

seanjohnston-storefeeder committed:
Merge branch 'goondev' into gooncrawl_vanilla-throwing-overhaul
